vnext/Shared/Networking/OriginPolicyHttpFilter.cpp

```diff
@@ -0,0 +1,638 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "OriginPolicyHttpFilter.h"
+
+// React Native Windows
+#include <RuntimeOptions.h>
+#include "WinRTTypes.h"
+
+// Boost Library
+#include <boost/algorithm/string.hpp>
+#include <boost/lexical_cast/try_lexical_convert.hpp>
+
+// Windows API
+#include <winrt/Windows.Web.Http.Headers.h>
+
+// Standard Library
+#include <regex>
+
+using std::set;
+using std::wstring;
+
+using winrt::hresult_error;
+using winrt::hstring;
+using winrt::to_hstring;
+using winrt::Windows::Foundation::IInspectable;
+using winrt::Windows::Foundation::Uri;
+using winrt::Windows::Web::Http::HttpMethod;
+using winrt::Windows::Web::Http::HttpRequestMessage;
+using winrt::Windows::Web::Http::HttpResponseMessage;
+using winrt::Windows::Web::Http::Filters::IHttpFilter;
+using winrt::Windows::Web::Http::Headers::HttpMediaTypeHeaderValue;
+using winrt::Windows::Web::Http::Headers::HttpRequestHeaderCollection;
+
+namespace Microsoft::React::Networking {
+
+#pragma region OriginPolicyHttpFilter
```

Comment:
Should we limmit the Scheme to http/https as well?

Reply:
I don't think we should in all cases.

That said, I did set up a runtime option for that. See `New runtime options` in the description.